src/N3Store.js

```diff
@@ -790,9 +820,139 @@ export default class N3Store {
       this.removeQuads(toRemove);
     return lists;
   }
+
+  // ### Store is an iterable.
+  // Can be used where iterables are expected: for...of loops, array spread operator,
+  // `yield*`, and destructuring assignment (order is not guaranteed).
+  *[Symbol.iterator]() {
+    yield* this.getQuads();
+  }
 }
 
 // Determines whether the argument is a string
 function isString(s) {
   return typeof s === 'string' || s instanceof String;
 }
+
+/**
+ * A class that implements both the DatasetCore interface and the Readable stream
+ * interface.
+ */
+class DatasetCoreAndReadableStream {
```

Comment:
Another way to implement this functionality is to instead of returning a hybrid object from match is to make the N3Store itself implement the Readable stream interface and return an instance of N3Store from match. That way, users can treat it like a Readable stream, a DatasetCore, or the full N3Store. Otherwise, only the 4 methods of DatasetCore are available plus the ones for Readable stream. Thoughts?